sdk/src/main/java/com/sygic/travel/sdk/auth/facade/AuthFacade.kt

```diff
@@ -0,0 +1,170 @@
+	) {
+		val request = RequestBuilder.passwordAuthRequest(username, password)
+		runWithCallback({ authService.authorize(request) }, callback)
+	}
+
+	suspend fun loginUserByPassword(username: String, password: String): UserSession? {
+		val request = RequestBuilder.passwordAuthRequest(username, password)
+		return runAsync { authService.authorize(request) }
+	}
+
+	fun loginUserByGoogle(googleToken: String?, callback: Callback<UserSession?>) {
+		val request = RequestBuilder.googleAuthRequest(googleToken)
+		runWithCallback({ authService.authorize(request) }, callback)
+	}
+
+	suspend fun loginUserByGoogle(googleToken: String?): UserSession? {
+		val request = RequestBuilder.googleAuthRequest(googleToken)
+		return runAsync { authService.authorize(request) }
+	}
+
+	fun loginUserByFacebook(facebookToken: String?, callback: Callback<UserSession?>) {
+		val request = RequestBuilder.facebookAuthRequest(facebookToken)
+		runWithCallback({ authService.authorize(request) }, callback)
+	}
+
+	suspend fun loginUserByFacebook(facebookToken: String?): UserSession? {
+		val request = RequestBuilder.facebookAuthRequest(facebookToken)
+		return runAsync { authService.authorize(request) }
+	}
+
+	fun loginUserByDeviceId(callback: Callback<UserSession?>) {
+		val authRequest = RequestBuilder.deviceIdAuthRequest(authStorageService.getDeviceId())
+		runWithCallback({ authService.authorize(authRequest) }, callback)
+	}
+
+	suspend fun loginUserByDeviceId(): UserSession? {
+		val authRequest = RequestBuilder.deviceIdAuthRequest(authStorageService.getDeviceId())
+		return runAsync { authService.authorize(authRequest) }
+	}
+
+	fun loginUserByJwtToken(jwtToken: String, callback: Callback<UserSession?>) {
+		val authRequest = RequestBuilder.jwtTokenAuthRequest(jwtToken)
+		runWithCallback({ authService.authorize(authRequest) }, callback)
+	}
+
+	suspend fun loginUserByJwtToken(jwtToken: String): UserSession? {
+		val authRequest = RequestBuilder.jwtTokenAuthRequest(jwtToken)
+		return runAsync { authService.authorize(authRequest) }
+	}
+
+	fun registerUser(
+		name: String,
+		email: String,
+		password: String,
+		callback: Callback<UserRegistrationResponse?>
+	) {
+		val userRegistrationRequest = RequestBuilder.userRegistrationRequest(email, password, name)
+		runWithCallback({ authService.register(userRegistrationRequest) }, callback)
+	}
+
+	suspend fun registerUser(
+		name: String,
+		email: String,
+		password: String
+	): UserRegistrationResponse? {
+		val userRegistrationRequest = RequestBuilder.userRegistrationRequest(email, password, name)
+		return runAsync { authService.register(userRegistrationRequest) }
+	}
+
+	fun logoutUser() {
+		authService.setAuthorizationStorage(null, 0, null)
+	}
+
+	fun getUserSession(callback: Callback<UserSession?>) {
+		runWithCallback({ authService.getUserSession() }, callback)
+	}
+
+	suspend fun getUserSession(): UserSession? {
+		return runAsync { authService.getUserSession() }
+	}
+
+	private object RequestBuilder {
```

Comment:
this can't be here. let's create a facade for this.
and maybe rename the direct accessible objects in SDK to managers? 
suggest something...

Reply:
I would put these methods directly into the service.